common/work.c

```diff
@@ -17,45 +17,88 @@ static struct {
 	void         *param;
 } work_items[MAX_WORK_ITEMS];
 
+static size_t queue_head = 0;
+static size_t queue_tail = MAX_WORK_ITEMS;
+
+/**
+ * Determines if the queue is full using the given arguments. The values of
```

Comment:
"Determines" → "Determine" (be imperative), and the same for the other doc blocks